packages/shared/lib/core/network/utils/checkNodeUrlValidity.ts

```diff
@@ -19,6 +19,14 @@ export function checkNodeUrlValidity(
         return 'error.node.invalid'
     }
 
+    // Check if it's HTTP and localhost
+    const isHttpLocalhost =
+        newUrl.toLowerCase().startsWith('http://localhost') || newUrl.toLowerCase().startsWith('http://127.0.0.1')
+    if (isHttpLocalhost && allowInsecure) {
```

Comment:
forcing `allowInsecure` in this check defeats the whole purpose of allowing a user connecting to http localhost, users will not have in production artifacts `allowInsecure` to true